File: jour_9/jour9.py
print("==========calcul de la moyenne et mention==========")
while True :
    try :
        a = float(input("Entrez la premiére note : "))
        b = float(input("Entrez la seconde note : "))
        c = float(input("Entrez la troisiéme note : "))
        break
    except ValueError :
        print("entrez invaide réessayer\n")
moyenne = (a + b + c)/3
if (moyenne < 10 and moyenne > 0) :
    print(f"Moyenne : {moyenne}\nMention : Ajournée.")
elif (moyenne >= 10 and moyenne < 12) : 
    print(f"Moyenne : {moyenne}\nMention : passable.")
elif (moyenne >= 12 and moyenne < 14) :
    print(f"Moyenne : {moyenne}\nMention : Assez-bien.")
elif (moyenne >= 14 and moyenne < 16) :
    print(f"Moyenne : {moyenne}\nMention : Bien.")
elif (moyenne >= 16 and moyenne < 19) : 
    print("Trés bien ")
elif (moyenne >= 19 and moyenne <= 20 ) : 
    print(f"Moyenne : {moyenne}\nMention : Excellent.")
else :
    print("Note invalide.")

File: jour_9/jour9.c
#include<stdio.h>
int main(void){
    float a, b, c;
    printf("==========calcul de la moyenne et mention==========\n");
    do{
        printf("Entrez la premiére note : ");
        scanf("%f",&a);
        printf("Entrez la deuxiéme note : ");
        scanf("%f",&b);
        printf("Entrez la troisiéme note : ");
        scanf("%f",&c);
        if(a < 0 || b < 0 || c < 0)
            printf("note invalide.veuillez réessayer\n");    
    }while(a < 0 || b < 0 || c < 0);
    float moyenne = (a + b + c)/3;
    if (moyenne < 10 && moyenne >= 0) 
        printf("Moyenne : %.2f\nMention : Ajournée.", moyenne);
    else if (moyenne >= 10 && moyenne < 12)  
        printf("Moyenne : %.2f\nMention : passable.",moyenne);
    else if (moyenne >= 12 && moyenne < 14) 
        printf("Moyenne : %.2f\nMention : Assez-bien.",moyenne);
    else if (moyenne >= 14 && moyenne < 16) 
        printf("Moyenne : %.2f\nMention : Bien.",moyenne);
    else if (moyenne >= 16 && moyenne < 19)  
        printf("Trés bien ");
    else if (moyenne >= 19 && moyenne <= 20 )  
        printf("Moyenne : %.2f\nMention : Excellent.",moyenne);
    else 
        printf("Note invalide.\n ");
}